examples/lightning_base.py

```diff
@@ -206,6 +234,12 @@ def add_model_specific_args(parser, root_dir):
             "--attention_dropout", type=float, help="Attention dropout probability (Optional). Goes into model.config",
         )
         parser.add_argument("--learning_rate", default=5e-5, type=float, help="The initial learning rate for Adam.")
+        parser.add_argument(
+            "--lr_scheduler",
+            default="linear",
+            type=str,
```

Comment:
if you pass `choices=["linear", "cosine", "cosine_with_restarts"]` I think you will get a nice traceback for `--lr_scheduler=help` without having to add an override the "help" value.

Reply:
yes, when we originally discussed this - that was your other suggestion and your said the =help override is preferable. But I will change it to this suggestion. Thank you.

Reply:
I second this, putting them in `choices` means that when doing `python script.py --help`, the choices will be shown for the `lr_scheduler` argument. This way there's no need to handle  the help and invalid scheduler option.

Reply:
Ah, one shortcoming of this approach is that it'll not be able to tell the user which actual scheduler will be run, just the alias.
Current code produces:
```
Available lr_schedulers:
--lr_scheduler=cosine (get_cosine_schedule_with_warmup)
--lr_scheduler=cosine_w_restarts (get_cosine_with_hard_restarts_schedule_with_warmup)
--lr_scheduler=linear (get_linear_schedule_with_warmup)
--lr_scheduler=help (this help)
```
choices will just list the keys. 

thoughts?

Reply:
I think the aliases are clear enough, no?

Reply:
Should there also be some sort of logger that optinally logs which scheduler is being run? Or is it unnecessary?

Reply:
ok, using `choices` results in an ugly output, e.g.:

```
                   [--learning_rate LEARNING_RATE]
                   [--lr_scheduler {aaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,cccaaa,bbb,ccc}]
                   [--weight_decay WEIGHT_DECAY] [--adam_epsilon ADAM_EPSILON]
```